mergify_engine/delayed_refresh.py

```diff
@@ -44,11 +45,18 @@ async def plan_next_refresh(match: rules.RulesEvaluator, ctxt: context.Context)
 
     if best_bet is None or best_bet >= filter.DT_MAX:
         await ctxt.redis.zrem(DELAYED_REFRESH_KEY, zset_subkey)
-    else:
-        await ctxt.redis.zadd(
-            DELAYED_REFRESH_KEY,
-            **{zset_subkey: best_bet.timestamp()},
-        )
+        return
+
+    in_ten_minutes = utils.utcnow() + datetime.timedelta(
+        minutes=DELAYED_REFRESH_MIN_MINUTES
+    )
+    if best_bet < in_ten_minutes:
+        best_bet = in_ten_minutes
```

Comment:
I'm not sure I get the optimization here.
It's 12:03.
If the next best bet is today at 12:10 because the condition is `>=12:10`.

Now you're saying that since 12:10 is < than (12:03+10min), we should reschedule at 12:13.

Alright, but what's the point? What does that optimize in the end?

You're still refreshing the PR

Worst than that, if the PR has the condition: `>=12:10` and `<=12:20` and it's 12:09, then you're going to schedule the refresh as 12:09+10min = 12:19, which gives a very narrow window of 1 minute to evaluate the rule to get it true.

I feel like I miss something?

Reply:
You feel I miss something! 

I just reread and this clearly didn't do what I expected. I wanted to ensure we didn't refresh a pull request every seconds/minutes but every 5 minutes minimum.